internal/stream/stream.go

```diff
@@ -358,6 +414,8 @@ func NewPushStream(ctx context.Context, nodeID string, subName string, subscript
 		subscriberCore:   subscriberCore,
 		doneCh:           make(chan struct{}),
 		httpClient:       httpclient,
+		restartChan:      make(chan bool, 10),
```

Comment:
Both channels are signal-based channels. Avoid buffered channels for this use case.

Reply:
Have make these unbuffered, but in case ctx is done, I am calling stop() directly rather than sending signal on stop chan, as the stream manager also exists when parent ctx is done